File: gymflow-backend/src/payment/payment.service.ts
import { Injectable, NotFoundException } from '@nestjs/common';
import { PrismaService } from '../prisma/prisma.service';
import { CreatePaymentDto } from './dto/create-payment.dto';

@Injectable()
export class PaymentService {
  constructor(private prisma: PrismaService) {}

  async create(createPaymentDto: CreatePaymentDto) {
    const member = await this.prisma.members.findUnique({
      where: { id: createPaymentDto.member_id },
    });
    if (!member) throw new NotFoundException('Member not found');

    return this.prisma.payments.create({
      data: {
        member_id: createPaymentDto.member_id,
        amount: createPaymentDto.amount,
        due_date: new Date(createPaymentDto.due_date),
        status: createPaymentDto.status,
        paid_date: createPaymentDto.paid_date ? new Date(createPaymentDto.paid_date) : null,
      },
    });
  }

  async findUnpaid() {
    return this.prisma.payments.findMany({
      where: { status: 'UNPAID' },
      include: {
        member: {
          select: { name: true, phone: true },
        },
      },
      orderBy: { due_date: 'asc' },
    });
  }

  async findHistoryByMember(memberId: number) {
    return this.prisma.payments.findMany({
      where: { member_id: memberId },
      orderBy: { due_date: 'desc' },
    });
  }
}

File: gymflow/lib/services/payment_service.dart
import 'api_service.dart';

class PaymentService {
  final ApiService _api = ApiService();

  Future<bool> recordPayment(int memberId, double amount, String dueDate, String status) async {
    try {
      await _api.client.post('/payments', data: {
        'member_id': memberId,
        'amount': amount,
        'due_date': dueDate,
        'status': status,
      });
      return true;
    } catch (e) {
      print('Failed to record payment: $e');
      return false;
    }
  }

  Future<List<dynamic>> getUnpaid() async {
    try {
      final response = await _api.client.get('/payments/unpaid');
      return response.data;
    } catch (e) {
      print('Failed to get unpaid: $e');
      return [];
    }
  }
}
